src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -986,6 +1003,33 @@ impl LinkCollector<'_, '_> {
                 }
             }
 
+            // Strip generics from the path.
+            if path_str.contains(['<', '>'].as_slice()) {
+                stripped_path_string = match strip_generics_from_path(path_str) {
+                    Ok(path) => path,
+                    Err(err_kind) => {
+                        debug!("link has malformed generics: {}", path_str);
+                        resolution_failure(
+                            self,
+                            &item,
+                            path_str,
+                            disambiguator,
+                            dox,
+                            link_range,
+                            smallvec![err_kind],
+                        );
+                        return;
+                    }
+                };
+                path_str = &stripped_path_string;
+            }
+
+            // The link is not an intra-doc link if it still contains angle brackets, commas,
+            // or spaces after stripping generics.
+            if path_str.contains(['<', '>', ',', ' '].as_slice()) {
```

Comment:
When would this ever contain `<` and not have a `MalformedGenerics` error?

Reply:
I guess I just figured I would add it as a sanity check. Should I remove it?

Reply:
If you want a sanity check, let's use an `assert` instead. I want to avoid silent failures (https://github.com/rust-lang/rust/pull/75756#discussion_r487213338).